profiles/views.py

```diff
@@ -175,19 +174,12 @@ def get_serializer_class(self):
         else:
             return ProfileOwnerDetailEditSerializer
 
-    def destroy(self, request, *args, **kwargs):
-        instance = self.get_object()
-        user = self.request.user
-        password = self.request.data.get("password")
-        if not password or not check_password(password, user.password):
-            return Response(status=status.HTTP_400_BAD_REQUEST)
-        else:
-            self.perform_destroy(instance)
-            return Response(status=status.HTTP_204_NO_CONTENT)
-
     def perform_destroy(self, instance):
-        instance.is_deleted = True
-        instance.save()
+        request_data = {"password": self.request.data.get("password")}
+        serializer = self.get_serializer(instance)
+        if serializer.validate_for_delete(request_data):
```

Comment:
Why do we manually call the validation method?

Write the validation method. Use it on serializer field. DRF will care about the rest.

Reply:
Fixed